src/NCronJob/Registry/JobRun.cs

```diff
@@ -124,7 +124,9 @@ private void SetState(JobState state)
 
     public void NotifyStateChange(JobStateType type, Exception? fault = default)
     {
-        if (CurrentState.Type == type || IsCompleted)
+        var stateIsUnchangedAndNotRetrying = CurrentState.Type == type && type != JobStateType.Retrying;
```

Comment:
Maybe: Can we put this as extension method onto `JobStateType`?

Something like:
```csharp
internal static class JobStateTypeExtensions
{
  public static bool IsUnchangedAndNotRetrying(this JobStateType type) ...
}
```

Reply:
On it

Reply:
Done. Eventually went with JobState extension methods than JobStateType ones.

@linkdotnet Does that fit you?